src/game/boe.actions.cpp

```diff
@@ -1590,12 +1590,39 @@ bool handle_keystroke(const sf::Event& event){
 	Key talk_chars[9] = {kb::L,kb::N,kb::J,kb::B,kb::S,kb::R,kb::D,kb::G,kb::A};
 	Key shop_chars[8] = {kb::A,kb::B,kb::C,kb::D,kb::E,kb::F,kb::G,kb::H};
 	
-	if(map_visible && event.key.code == kb::Escape
-	   && (overall_mode != MODE_TALKING) && (overall_mode != MODE_SHOPPING)) {
-		mini_map.setVisible(false);
-		map_visible = false;
-		mainPtr.setActive();
-		return false;
+	if(event.key.code == kb::Escape) {
+		bool abort=true;
```

Comment:
I think this is broadly a good idea, but I'm wondering if you might have missed a detail somewhere. Most of these modes have another way to cancel them, don't they? Usually by pressing the same key again? Does cancelling with Escape have the exact same outcome as cancelling normally?

I was trying to determine this for myself and found something that suggested it might not be the case for at least `MODE_TOWN_TARGET`, but I didn't look much further beyond that.

Ultimately, this is a good idea but probably not a good time to do it. We need to refactor the input so that the action and the input that triggers it aren't so tightly coupled. Then implementing this would probably be much easier to do correctly.

Reply:
Yes, you can always cancel an action by pressing the same key again. But in practice, when I want to undo an action, I find it easier to always press the Escape key than to have to search for the key I need to press to undo that action.